planner/core/find_best_task.go

```diff
@@ -282,6 +282,11 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 		if curTask.cost() < bestTask.cost() || (bestTask.invalid() && !curTask.invalid()) {
 			bestTask = curTask
 		}
+
+		//// TODO: introduce a new variable as a switch to control this
+		//if curTask.plan().CalPlanCost(property.RootTaskType) < bestTask.plan().CalPlanCost(property.RootTaskType) || (bestTask.invalid() && !curTask.invalid()) {
```

Comment:
It's possible that the task is not root type right?
Such as the case that double read case

Reply:
You are right. Now I introduce a new function `getTaskType(task) type` to get their types.